packages/backend-lib/src/destinations/amazonses.ts

```diff
@@ -0,0 +1,199 @@
+import Https from "node:https";
+
+import {
+  MessageTag,
+  SendEmailCommand,
+  type SendEmailCommandInput,
+  SendEmailCommandOutput,
+  SESClient,
+  SESServiceException,
+} from "@aws-sdk/client-ses";
+import { err, Result, ResultAsync } from "neverthrow";
+import SnsPayloadValidator from "sns-payload-validator";
+
+import { submitBatch } from "../apps";
+import logger from "../logger";
+import {
+  AmazonSesConfig,
+  AmazonSesEventPayload,
+  AmazonSesMailFields,
+  AmazonSesNotificationType,
+  AmazonSNSEvent,
+  AmazonSNSSubscriptionEvent,
+  AmazonSNSUnsubscribeEvent,
+  BatchTrackData,
+  EventType,
+  InternalEventType,
+} from "../types";
+
+function unwrapTag(tagName: string, tags: Record<string, string[]>) {
+  if (!tags[tagName]) {
+    return null;
+  }
+  // disable the nonnull assertion eslint rule
+  // eslint-disable-next-line
+  const tag = tags[tagName]!;
```

Comment:
rather than asserting non-null we can do something like `return tags[tagName]?[0] ?? null`

Reply:
Yep, for some reason TS was being grumpy about that syntax last night... I'll take another look and see what's going on.

Reply:
This is totally what was tripping me up, I thought the syntax was what you suggested as well.  Turns out, its actually `tags[tagName]?.[0]`.  Grr..

Reply:
Perfect! That syntax is counterintuitive which must be why i forgot it!